system/build.sh

```diff
@@ -0,0 +1,178 @@
+#!/bin/bash
+
+# Make sure we've got the compiler
+if [ "`which arm-none-eabi-gcc`" == "" ]; then
+  if [ "`which apt`" != "" ]; then
+    echo "Installing gcc-arm-none-eabi..."
+    sudo apt install -y gcc-arm-none-eabi
+  else
+    echo "[ERROR] arm-none-eabi-gcc not found!"
+    exit 1
+  fi
+fi
+
+# Required device
+DEVICE="$1"
+if [ "$DEVICE" == "" ]; then
+  echo
+  echo "Usage: $0 [DEVICE] [THREADING]"
+  echo
+  echo " DEVICE"
+  echo "   ds203: Build for DS203"
+  echo "   ds213: Build for DS213"
+  echo "   la104: Build for LA104"
+  echo "   all:   Build for all the devices"
+  echo
+  echo " THREADING"
+  echo "   single: Build one app at a time (Default)"
+  echo "   multi:  Build all the apps in a dir at the same time"
+  echo "   all:    Build ALL the apps at the same time"
+  echo
+  exit 1
+fi
+
```

Comment:
Could you please add some special switch or attach this script to the docker? There is build.sh script in the root which is run through docker. And I would like to have only single building script.
The task of the docker script is tu build the firmwares and some sample apps. It would be great, if we could extend it so it builds the firmwares and disk images (currently only release_image.sh does this for LA104, but others will be added soon). In my case whole building takes less than 2 minutes and I use it only when I make some significant changes to the OS API, so I am not sure if reducing the building time is necessary, but at least we can make building process simpler for others and have just one single building script that does all the stuff for us (currently there are two and this PR would add another one)

Reply:
Makes sense. I'll see what I can come up with.

Reply:
...just few more details about the docker building script:
When everything worked fine, it should create folder /output in the root and place there firmwares (system_ds203.hex, system_ds213.hex, system_la104.hex) and the disk images (currently only la104apps.zip from system/release). Those files are the result of building whole repository and I believe we can somehow automatize this to produce releases in more frequent way

Reply:
Hmm... Now that I think about it, from my previous git experience, committing builds to git is comfy but at some point the repo will get quite big. It's probably not an issue yet because the binaries are rather small and releases are rare. But this can quickly become an issue if releases are done more often. Git can't do any diff on binaries to store smaller patches so it stores the entire elf file.

I actually based my OTA update for [pocketbook-coolreader](https://github.com/blchinezu/pocketbook-coolreader) on storing the builds in github and the repo size reached 1.6GB. I figured it out when it was too late.

Maybe it would be better to have the script build all there is to build and publish the releases under the releases section of github. Not sure if/how we can automatically publish releases this way though. Might require manual upload.

--------------------------------------

Assuming we keep the binaries in the git repo though, if we put the releases in `/output` then we should add `/system/releases` in `.gitignore`. It would be pointless to have the same binary stored twice.

Reply:
We should use the github CI to build and publish releases. We can do that using a combination of a custom flow see .."http://github/thisrepo/actions/new" the instructions given in the docker section of the readme and the github publish release.

Reply:
I do not want to store builds in repository. I am considering of removing whole system/release folder and building github releases in regular basis through github's CI. I know it is not a good practise to keep binaries in then repository. It was meant as a temporary solution for easier distribution of the firmware and applications until the projects becomes more stable. And since @mgazza helped us to integrate CI/docker, it would be good idea to integrate these building scripts with it - In his script, it just creates /output folder and places there some freshly built binaries. This folder is not being pushed into the repository in any way - but I guess these building artefacts can be later used for publishing releases / nightly builds. I will try to learn more about the CI/releases during the weekend, but now I do not have much time for that.

Reply:
@gabonator if you need any help hit me up on telegram :)